tests/utilities/test_fetching.py

```diff
@@ -176,6 +180,14 @@ def test_dataloader(self):
 def test_trainer_num_prefetch_batches(tmpdir):
 
     model = RecommenderModel()
+
+    class CheckDataFetcher(Callback):
+        def __init__(self, data_fetcher_cls: Type):
+            self.data_fetcher_cls = data_fetcher_cls
+
+        def on_batch_start(self, trainer, *_) -> None:
+            assert isinstance(trainer.data_connector.train_data_fetcher, self.data_fetcher_cls)
```

Comment:
note we are relying on the callback implementation for testing here. this is fragile